unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -773,23 +773,36 @@ loop e = do
 
                 fixupOutput :: Path.HQSplit -> HQ.HashQualified Name
                 fixupOutput = fmap Path.unsafeToName . HQ'.toHQ . Path.unsplitHQ
-            NamesI global thing -> do
+            NamesI global query -> do
               Cli.Env {codebase} <- ask
+              currentPath' <- Path.unabsolute <$> Cli.getCurrentPath
               hqLength <- liftIO (Codebase.hashLength codebase)
-              basicPrettyPrintNames <- getBasicPrettyPrintNames
-              ns0 <- if global then pure basicPrettyPrintNames else basicParseNames
-              let ns = NamesWithHistory ns0 mempty
-                  terms = NamesWithHistory.lookupHQTerm thing ns
-                  types = NamesWithHistory.lookupHQType thing ns
-                  printNames = NamesWithHistory basicPrettyPrintNames mempty
+              root <- Cli.getRootBranch
+              (names, pped) <-
+                if global || any Name.isAbsolute query
```

Comment:
Currently `names` only accepts a single argument; the `any` is for folding out of the `HashQualified` wrapper, since that may or may not contain a name and we want a `False` result if it doesn't have one.